pkg/controller/csr_check.go

```diff
@@ -159,44 +162,202 @@ func authorizeCSR(
 		return false, nil
 	}
 
+	var approvalErrors []error
+
 	// Check for an existing serving cert from the node.  If found, use the
 	// renewal flow.  Any error connecting to the node, including validation of
 	// the presented cert against the current Kubelet CA, will result in
 	// fallback to the original flow relying on the machine-api.
 	//
 	// This is only supported if we were given a CA to verify against.
+	var servingCert *x509.Certificate
 	if ca != nil {
-		servingCert, err := getServingCert(c, nodeAsking, ca)
-		if err == nil && servingCert != nil {
-			klog.Infof("Found existing serving cert for %s", nodeAsking)
+		var err error
+		servingCert, err = getServingCert(c, nodeAsking, ca)
+		if err != nil {
+			klog.Infof("Failed to retrieve current serving cert: %v", err)
+		}
+	}
 
-			err := authorizeServingRenewal(nodeAsking, csr, servingCert, x509.VerifyOptions{Roots: ca})
+	x509VerificationOpts := x509.VerifyOptions{Roots: ca}
+	if servingCert != nil {
+		klog.Infof("Found existing serving cert for %s", nodeAsking)
 
-			// No error, the renewal is authorized.
-			if err == nil {
-				return true, nil
-			}
+		err := authorizeServingRenewal(nodeAsking, csr, servingCert, x509VerificationOpts)
 
-			klog.Infof("Could not use current serving cert for renewal: %v", err)
-			klog.Infof("Current SAN Values: %v, CSR SAN Values: %v",
-				certSANs(servingCert), csrSANs(csr))
+		// No error, the renewal is authorized.
+		if err == nil {
+			return true, nil
 		}
 
-		if err != nil {
-			klog.Infof("Failed to retrieve current serving cert: %v", err)
-		}
+		approvalErrors = append(approvalErrors, err)
+		klog.Infof("Could not use current serving cert for renewal: %v", err)
+		klog.Infof("Current SAN Values: %v, CSR SAN Values: %v",
+			certSANs(servingCert), csrSANs(csr))
 	}
 
 	// Fall back to the original machine-api based authorization scheme.
 	klog.Infof("Falling back to machine-api authorization for %s", nodeAsking)
+	if err := authorizeServingCertWithMachine(machines, req, nodeAsking, csr); err != nil {
+		approvalErrors = append(approvalErrors, err)
+		klog.Infof("Could not use Machine for serving cert authorization: %v", err)
+	} else {
+		// No error means the machine was able to authorize the cert
+		return true, nil
+	}
+
+	network := &configv1.Network{}
+	if err := c.Get(context.Background(), client.ObjectKey{Name: "cluster"}, network); err != nil {
```

Comment:
can we move "cluster" and "OpenShiftSDN" into constants then move the `get configv1.Network{}` and `network.Status.NetworkType == "OpenShiftSDN"` logic into a unit tested function `isEgress(node)` that it also accounts for https://docs.openshift.com/container-platform/4.6/networking/ovn_kubernetes_network_provider/configuring-egress-ips-ovn.html#nw-egress-ips-node-assignment_configuring-egress-ips-ovn into the discrimitation criteria?